usd/include/sdf/usd/usd_parser/USDTransforms.hh

```diff
@@ -0,0 +1,146 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef SDF_USD_USD_PARSER_UTILS_HH_
+#define SDF_USD_USD_PARSER_UTILS_HH_
+
+#include <string>
+#include <vector>
+
+#include <ignition/math/Pose3.hh>
+#include <ignition/math/Quaternion.hh>
+#include <ignition/math/Vector3.hh>
+
+#include <ignition/utils/ImplPtr.hh>
+
+// TODO(ahcorde) this is to remove deprecated "warnings" in usd, these warnings
+// are reported using #pragma message so normal diagnostic flags cannot remove
+// them. This workaround requires this block to be used whenever usd is
+// included.
+#pragma push_macro ("__DEPRECATED")
+#undef __DEPRECATED
+#include <pxr/usd/usdGeom/gprim.h>
+#pragma pop_macro ("__DEPRECATED")
+
+#include "sdf/sdf_config.h"
+#include "sdf/system_util.hh"
+#include "sdf/usd/Export.hh"
+#include "sdf/usd/UsdError.hh"
+
+#include "USDData.hh"
+
+namespace sdf
+{
+  // Inline bracke to help doxygen filtering.
+  inline namespace SDF_VERSION_NAMESPACE {
+  //
+  namespace usd
+  {
+    /// \brief This class stores the transforms of a schema
+    /// This might contain scale, translate or rotation operations
+    /// The booleans are used to check if there is a transform defined
+    /// in the schema
+    /// Rotation is splitted in a vector because this might be defined
+    /// as a rotation of 3 angles (ZYX, XYZ, etc).
+    class IGNITION_SDFORMAT_USD_VISIBLE UDSTransforms
+    {
+      /// \brief Default constructor
+      public: UDSTransforms();
+
+      /// \brief Translate
+      public: ignition::math::Vector3d Translate();
+
+      /// \brief Scale
+      public: ignition::math::Vector3d Scale();
+
+      /// \brief Rotation
+      public: std::vector<ignition::math::Quaterniond> Rotations();
+
+      /// \brief Set translate
+      public: void SetTranslate(const ignition::math::Vector3d &_translate);
+
+      /// \brief Set scale
+      public: void SetScale(const ignition::math::Vector3d &_scale);
+
+      /// \brief Add rotation
+      public: void AddRotation(const ignition::math::Quaterniond &_q);
+
+      /// \brief True if there is a rotation ZYX defined or false otherwise
+      bool RotationZYX();
+
+      /// \brief True if there is a rotation XYZ defined or false otherwise
+      bool RotationXYZ();
+
+      /// \brief True if there is a rotation (as a quaterion) defined
+      /// or false otherwise
+      bool Rotation();
+
+      /// \brief Set if there is any rotation ZYX defined
+      void SetRotationZYX(bool _rotationZYX);
+
+      /// \brief Set if there is any rotation XYZ defined
+      void SetRotationXYZ(bool _rotationXYZ);
+
+      /// \brief Set if there is any rotation defined
+      void SetRotation(bool _rotation);
```

Comment:
A few things about these methods:
* I believe that our style guide says to avoid using the `Set` prefix for setters
* can documentation be added for the input parameters?
* It would be nice to add documentation that outlines the difference between `Rotation` and `RotationZYX`, and the difference between `Rotation` and `RotationXYZ`. For example, is could `Rotation` be `false` and `RotationXYZ`/`RotationZYX` be `true`? Or is `Rotation` always true if `RotationXYZ` or `RotationZYX` is true?